src/librustc/mir/repr.rs

```diff
@@ -216,6 +213,13 @@ impl BasicBlock {
         BasicBlock(index as u32)
     }
 
+    /// Returns a BasicBlock with index 1. This is actual end block (containing
+    /// the Return terminator) only during the building of MIR and should not be
+    /// used outside that.
+    pub const fn end_block() -> BasicBlock {
+        BasicBlock(1)
+    }
+
```

Comment:
Why have this at all? Why not make it somehow local to the `build` package? Ideally, it'd just be a field of the `Builder`.